src/app/slash_plugins/ta.plugin.ts

```diff
@@ -0,0 +1,259 @@
+import {
+  CommandInteraction,
+  GuildMember,
+  MessageEmbed,
+  Snowflake,
+  TextChannel,
+  User,
+} from 'discord.js';
+import mongoose, { Document } from 'mongoose';
+import Constants from '../../common/constants';
+import { Command } from '../../common/slash';
+import { IContainer } from '../../common/types';
+import { ClassTAModel } from '../../schemas/class.schema';
+
+export interface ITAEntry {
+  userID: Snowflake;
+  chanID: Snowflake;
+  guildID: Snowflake;
+}
+
+export type TADocument = ITAEntry & Document;
+
+const ALLOWED_ROLES = [Constants.Roles.TeachingAssistant, Constants.Roles.Professor];
```

Comment:
This gating should be configured by the server admins and not the developers. The admins can go into the integrations and set roles that are and aren't allowed to use this rather than it being hardcoded.

For the time being, we can just set the `defaultMemberPermissions` to `'ADMIN'` so no one untrusted can use the command in the meantime.

Ideally, in the future we don't have to hardcode roles at all.